staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1134,16 +1144,16 @@ func mergePatchIntoOriginal(original, patch map[string]interface{}, t reflect.Ty
 				return mergepatch.ErrBadArgType(patchFieldValue, patchList)
 			}
 		}
-		fieldType, patchStrategies, mergeKey, err = forkedjson.LookupPatchMetadata(t, originalKey)
+		subschema, patchMeta, err = schema.LookupPatchMetadataForStruct(originalKey)
```

Comment:
at this point, we know the field referenced by `originalKey` is a slice, right? This is where I expected to see `schema.LookupPatchMetadataForStruct(originalKey) elemTypeSchema, patchMeta, error`

Reply:
then the `subschema, patchStrategy, patchMeta` passed to `mergeSliceHandler` already have the data needed by that method... the schema of the elements, the strategy and patch metadata for the slice

Reply:
> at this point, we know the field referenced by originalKey is a slice, right?

Yes.
Please check the comment for this function:
```
// mergePatchIntoOriginal processes $setElementOrder list.
```

Reply:
sorry, I meant `LookupPatchMetadataForSlice(originalKey) elemTypeSchema, patchMeta, error`

Reply:
```
func mergePatchIntoOriginal(original, patch map[string]interface{}, schema LookupPatchMeta, mergeOptions MergeOptions) error 
```
`original, patch` are `map[string]interface{}`, which is the parent of `$setElementOrder` list and its corresponding parallel list. But `original[originalKey]` is a list.

So at this level we still need to use `LookupPatchMetadataForStruct(originalKey)`.

Reply:
I expected the behavior you had in the original function: given the name of a list field, return the schema of the list element type, and the patch strategy and meta of the list field. I just wanted it to be explicit that the lookup was for a list.  

Hence: `LookupPatchMetadataForSlice(originalKey) elemTypeSchema, patchMeta, error`

Reply:
I understood what you want now.
```
func (schema) LookupPatchMetadataForSlice(key string) {
  fieldSchema, patchMeta, err := schema.LookupPatchMetadataForStruct(key)
  subschema := schema.getElemTypeSchema()
  return subschema, patchMeta, err
}
```
I will update the PR before Monday.